builder/proxmox/common/builder.go

```diff
@@ -47,29 +47,33 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook,
 
 	comm := &b.config.Comm
 
-	// Build the steps
-	coreSteps := []multistep.Step{
-		&stepStartVM{
-			vmCreator: b.vmCreator,
-		},
-		commonsteps.HTTPServerFromHTTPConfig(&b.config.HTTPConfig),
-		&stepTypeBootCommand{
-			BootConfig: b.config.BootConfig,
-			Ctx:        b.config.Ctx,
-		},
-		&communicator.StepConnect{
-			Config:    comm,
-			Host:      commHost((*comm).Host()),
-			SSHConfig: (*comm).SSHConfigFunc(),
-		},
-		&commonsteps.StepProvision{},
-		&commonsteps.StepCleanupTempKeys{
-			Comm: &b.config.Comm,
-		},
-		&stepConvertToTemplate{},
-		&stepFinalizeTemplateConfig{},
-		&stepSuccess{},
+	// Build the steps (Order matters)
+	coreSteps := []multistep.Step{}
+
+	coreSteps = append(coreSteps, &stepStartVM{vmCreator: b.vmCreator})
+	coreSteps = append(coreSteps, commonsteps.HTTPServerFromHTTPConfig(&b.config.HTTPConfig))
+	coreSteps = append(coreSteps, &stepTypeBootCommand{
+		BootConfig: b.config.BootConfig,
+		Ctx:        b.config.Ctx,
+	})
+	coreSteps = append(coreSteps, &communicator.StepConnect{
+		Config:    comm,
+		Host:      commHost((*comm).Host()),
+		SSHConfig: (*comm).SSHConfigFunc(),
+	})
+	coreSteps = append(coreSteps, &commonsteps.StepProvision{})
+	coreSteps = append(coreSteps, &commonsteps.StepCleanupTempKeys{
+		Comm: &b.config.Comm,
+	})
+	coreSteps = append(coreSteps, &stepStopVM{})
+
+	if b.config.ConvertToTemplate {
+		coreSteps = append(coreSteps, &stepConvertToTemplate{})
```

Comment:
I like better when we pass the configuration to the step and within the step is determined if it should be skipped or not based on the configuration. It makes it easier to log "Skipping ..." and keeps this step slice clean.